peepingtom/utils/geometric_primitives.py

```diff
@@ -0,0 +1,201 @@
+import numpy as np
+
+from .components import Child, ArrayContainer
+
+
+class GeometricPrimitive(Child):
+    """
+    Base class for geometric primitive objects
+    """
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent)
+
+    def _min_attr(self, attribute_name):
+        attr = getattr(self, attribute_name)
+        return np.min(attr)
+
+    def _max_attr(self, attribute_name):
+        attr = getattr(self, attribute_name)
+        return np.max(attr)
+
+
+class Point(GeometricPrimitive):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @property
+    def _min_x(self):
+        return self._min_attr('x')
+
+    @property
+    def _min_y(self):
+        return self._min_attr('y')
+
+    @property
+    def _min_z(self):
+        return self._min_attr('z')
+
+    @property
+    def _max_x(self):
+        return self._max_attr('x')
+
+    @property
+    def _max_y(self):
+        return self._max_attr('y')
+
+    @property
+    def _max_z(self):
+        return self._max_attr('z')
+
+    @property
+    def xy(self):
+        return np.column_stack((self.x, self.y))
+
+    @property
+    def yx(self):
+        return np.column_stack((self.y, self.x))
+
+    @property
+    def xyz(self):
+        return np.column_stack((self.x, self.y, self.z))
+
+    @property
+    def zyx(self):
+        return np.column_stack((self.z, self.y, self.x))
```

Comment:
These refer to `self.x|y|z`, which are never instantiated in this class. Is this desired? I see that some subclasses have some of these attributes, but then maybe these should be there? If they stay here, then this base class should return `NotImplemented` from `self.x|y|z`.
In general, I feel like all these properties are a bit unnecessary. Why not just a method that accepts a string like `xyz` or `yx` and returns the appropriate stuff? I don't like much all this code duplication...